src/sentry/static/sentry/app/components/actions/resolve.jsx

```diff
@@ -0,0 +1,198 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {getShortVersion} from '../../utils';
+import {t} from '../../locale';
+import CustomResolutionModal from '../customResolutionModal';
+import MenuItem from '../menuItem';
+import DropdownLink from '../dropdownLink';
+import ActionLink from './actionLink';
+
+export default class ResolveActions extends React.Component {
+  static propTypes = {
+    hasRelease: PropTypes.bool.isRequired,
+    latestRelease: PropTypes.object,
+    onUpdate: PropTypes.func.isRequired,
+    orgId: PropTypes.string.isRequired,
+    projectId: PropTypes.string.isRequired,
+    shouldConfirm: PropTypes.bool,
+    confirmMessage: PropTypes.node,
+    disabled: PropTypes.bool,
+    isResolved: PropTypes.bool,
+    isAutoResolved: PropTypes.bool,
+    confirmLabel: PropTypes.string,
+  };
+
+  static defaultProps = {
+    isResolved: false,
+    isAutoResolved: false,
+    confirmLabel: 'Resolve',
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {modal: false};
+  }
+
+  onCustomResolution(statusDetails) {
+    this.setState({
+      modal: false,
+    });
+    this.props.onUpdate({
+      status: 'resolved',
+      statusDetails,
+    });
+  }
+
+  getButtonClass() {
+    return 'btn btn-default btn-sm';
+  }
+
+  renderResolved() {
+    let {isAutoResolved, onUpdate} = this.props;
+    let buttonClass = `${this.getButtonClass()} active`;
```

Comment:
again, would be nice for this to use classnames since we're refactoring it anyway